### PR TITLE
Update User Guide for newly added features

### PR DESCRIPTION
**Pull request by laihuiqi**
- Added table of contents
- Divided the features into types : Main features and Side features
- Add user guide to Main features :  `delete`, `clear`, `clear_by`, `revert`, `revert_all` and `exit`
- Add user guide to Side features : `list_task`, `find_task`, `list_todo`, `add_todo`, `edit_deadline`, `edit_content`, `delete_todo`, `clear_todo`, `list_note`, `add_note`, `delete_note` and `clear_note`
- Update code summary for respective commands

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+* If there is either no todo or note, it will show `No todo at the moment` or `No note at the moment` respectively. The other list will be displayed normally.
+
+### Search for a task (todo and notes) : `find_task`
+
+Searches the recorded lists of todos and notes by keyword (company name in todos and note content in notes)
+
+Format: `find_task KEYWORD`
+
+Searches for the todos or notes with the specified `KEYWORD`.
+The keyword refers to the company name in todos ot the note content in notes that the user intends to look for.
```
Comment (jianminglok) [inline docs/UserGuide.md:255]: I think there is a minor typo here, should be "todos or the note content"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+
+Examples:
+`find_task test` searches for all todos with `COMPANY_NAME` or all notes with `NOTE_CONTENT` as test.
+
+### Display a list of todo internship applications :`list_todo`
```
Comment (jianminglok) [inline docs/UserGuide.md:260]: I think it's better to have a space after ":" and before `list_todo` to make the whole document more unified

File: docs/UserGuide.md [hunk 3]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+Examples:
+
+* `list_todo` shows all the todo applications that the user has recorded.
+* If there are no todo applications for at the moment, `No todo at the moment` will be shown.
+
+### Add a todo application: `add_todo`
+
+Adds a todo internship application to the todo list
+
+Format: ` add_todo n/COMPANY_NAME j/JOB_TITLE by/DEADLINE`
```
Comment (jianminglok) [inline docs/UserGuide.md:275]: I think there's an extra space before add_todo

File: docs/UserGuide.md [hunk 4]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+- `DEADLINE` should be in the format yyyy-mm-dd.
+- `DEADLINE` should not be earlier than the date when the todo application is created.
+
+Examples:
+* `add n/Facebook j/Product Manager by/2023-06-07` adds a todo application for the Product Manager role at Facebook. The internship should be applied by 7 June 2023.
+* `add n/LinkedIn j/Software Engineer by/2023-10-04` adds a todo application for the Software Engineer role at LinkedIn.The internship should be applied by 4 October 2023.
```
Comment (jianminglok) [inline docs/UserGuide.md:281]: I believe the command here should be `add_todo`

File: docs/UserGuide.md [hunk 5]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+
+Format: `edit_content INDEX c/NOTE_CONTENT`
+- Edits the note content of the specified `INDEX` to the specified `NOTE_CONTENT`.
+- The index refers to the index number shown in the displayed todo list.
+- The index must be a positive integer 1, 2, 3, …​
+- **Note content is an optional field for todo applications**
+- Note content is empty (null) in default.
+
+Examples:
+* `edit_content 2 c/Venue changed` Changes the status of the 2nd todo application in the todo list to `Venue changed`.
```
Comment (jianminglok) [inline docs/UserGuide.md:310]: I believe it should be "Change the note content of the 2nd todo application"

File: docs/UserGuide.md [hunk 6]
```diff
@@ -168,12 +206,169 @@ The keyword refers to the status, role or company that the user intends to look
...
+* `delete_todo 2` Deletes the 2nd todo application in the todo list.
+
+### Clear all todo application entries : `clear_todo`
+
+Clears all todo application entries from the todo applications list
+
+Format: `clear_todo`
+**Note that the action is irreversible**
+
+### Display list of short note :`list_note`
```
Comment (jianminglok) [inline docs/UserGuide.md:333]: I think it's better to have a space after ":" and before `list_note` make the whole document more unified

**Review by jianminglok — COMMENTED** [review]
Thank you so much for updating the tables for contents for our user guide. I can see the changes have been merged in another PR. Overall the quality written is very good, but here's just some minor things that I think should be updated

**Comment by laihuiqi** [comment]
Thank you for your suggestion! The UG looks better now!

**Pull request merged**